sonar-html-plugin/src/main/java/org/sonar/plugins/html/checks/sonar/ImgWithoutAltCheck.java

```diff
@@ -26,7 +27,8 @@
 
   @Override
   public void startElement(TagNode node) {
-    if ((isImgTag(node) || isImageInput(node)) && !node.hasProperty("ALT")) {
+    if ((isImgTag(node) && !hasAltAttribute(node)) ||
+       ((isImageInput(node) || isAreaTag(node)) && !hasNonEmptyAltAttribute(node))) {
```

Comment:
`!hasNonEmptyAltAttribute(node)` is a double negation, once with the operator and once in the name. It'd be easier to read something like `hasInvalidAltAttribute`